### PR TITLE
Modify the Python 3 stubs for requests to be Python 2-compatible

### PR DESCRIPTION
**Pull request by JelleZijlstra**
This works towards fixing #924. To make the changes easier to review, I'm planning to
fix the issue in two steps:
- This commit makes the 3 stubs compatible with py2
- The next PR will move the 3 stubs to 2and3 and remove the 2 stubs

I wrote this code by diffing the files in third_party/{2,3}/requests and making
appropriate changes to the Python 3 stubs (mostly str/Text stuff). I verified
that the 3 stubs now pass mypy when parsed as 2.7, but I don't have an annotated
requests-using codebase to test on.

### CODE REVIEW

**Comment by ambv** [comment]
LGTM, I appreciate you splitting the changes that way. Makes for an easy review! I can see the value of keeping the 2 and 3 stubs together and the `requests` case is decently simple.

In general, I see you pushing towards 2to3. This is good, makes the experience more consistent. I trust you to weigh when the string type differences or `sys.version_info` branching makes it too hard to follow what's going on. We want to keep stubs separate in those cases. It's easier for users to debug their typing problems that way. We're also less prone to introduce bad types in stubs if there's less unionizing and branching.

Summing up, the general move to 2to3 is good as long as it doesn't make the resulting stubs overly complicated.

**Comment by JelleZijlstra** [comment]
Thanks, I generally agree. I think one of the best reasons for moving towards 2and3 is to prevent the stubs from diverging—it's easy for small code differences to creep into the stubs that don't reflect differences in behavior between different versions. For third-party libraries specifically, I expect it almost always makes sense to merge the stubs, since third-party libraries will usually use a single-source codebase that supports both Python 2 and 3 and attempt to deliver a consistent experience. Standard library modules are more likely to have diverged significantly.
